packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/rest_api/FlutterRestApi.kt

```diff
@@ -58,14 +59,25 @@ class FlutterRestApi {
                 return
             }
 
+            // Protect against empty bodies, causing Okhttp3 to crash. ByteArray must be non-empty
+            // so that AppSync SDK does not swap it for null.
+            if (HttpMethod.requiresRequestBody(methodName.toUpperCase(Locale.ROOT)) && !options.hasData()) {
```

Comment:
a few questions about this part and related logic:
* I have an open PR for this bug, but in a different way https://github.com/aws-amplify/amplify-flutter/pull/661 though that one can be closed, my bad for not self-assigning issue. 
* Do we want to provide a blank string for the user as the response body, only in Android? It may be unlikely for most folks, but adding a body to their request without them knowing could have some unintended outcomes, depending on their backend implementation. Sure, we want to prevent a fatal error (current behavior), but another possibly is a non-fatal exception so they know they have to fix the body on their end. In most cases, I would expect these requests to require bodies, although HTTP itself does not require it.
* nice part here is `requiresRequestBody`. I did not know about that and it's useful here.

Reply:
I'm open to either. iOS does not throw on an empty body for these methods, though. This seems like an Okhttp3 thing, only.

Reply:
You are right only an okhttp3 (thus android) thing, but I do not believe ios adds bodies to requests like this (correct me if I'm wrong) so adding it only in android could just make it more opaque and unexpected. You have a point though, that any android solutions adds to platform discrepancy, even if it makes it less bad. Another option would be to add a default body in dart so that it avoids the error and makes it consistent on platforms.